aquadoggo/src/db/stores/entry.rs

```diff
@@ -435,259 +431,247 @@ mod tests {
-        key_pair: KeyPair,
-        entry: Entry,
-        #[from(test_db)]
-        #[future]
-        db: TestSqlStore,
-    ) {
-        let db = db.await;
-        let entry_encoded = sign_and_encode(&entry, &key_pair).unwrap();
-        let operation_encoded = OperationEncoded::try_from(entry.operation().unwrap()).unwrap();
-        let doggo_entry = StorageEntry::new(&entry_encoded, &operation_encoded).unwrap();
-        let result = db.store.insert_entry(doggo_entry).await;
-
-        assert!(result.is_ok())
+    fn insert_entry(key_pair: KeyPair, entry: Entry, #[from(test_db)] runner: TestDatabaseRunner) {
+        runner.with_db_teardown(|db: TestDatabase| async move {
+            let entry_encoded = sign_and_encode(&entry, &key_pair).unwrap();
+            let operation_encoded = OperationEncoded::try_from(entry.operation().unwrap()).unwrap();
+            let doggo_entry = StorageEntry::new(&entry_encoded, &operation_encoded).unwrap();
+            let result = db.store.insert_entry(doggo_entry).await;
+
+            assert!(result.is_ok());
+        });
     }
 
     #[rstest]
-    #[tokio::test]
-    async fn try_insert_non_unique_entry(
+    fn try_insert_non_unique_entry(
         #[from(test_db)]
         #[with(10, 1)]
-        #[future]
-        db: TestSqlStore,
+        runner: TestDatabaseRunner,
     ) {
-        let db = db.await;
-        let author = Author::try_from(db.key_pairs[0].public_key().to_owned()).unwrap();
-        let log_id = LogId::new(1);
+        runner.with_db_teardown(|db: TestDatabase| async move {
+            let author = Author::try_from(db.key_pairs[0].public_key().to_owned()).unwrap();
+            let log_id = LogId::new(1);
 
-        let first_entry = db
-            .store
-            .get_entry_at_seq_num(&author, &log_id, &SeqNum::new(1).unwrap())
-            .await
-            .unwrap()
-            .unwrap();
+            let first_entry = db
+                .store
+                .get_entry_at_seq_num(&author, &log_id, &SeqNum::new(1).unwrap())
+                .await
+                .unwrap()
+                .unwrap();
 
-        let duplicate_doggo_entry = StorageEntry::new(
-            first_entry.entry_signed(),
-            first_entry.operation_encoded().unwrap(),
-        )
-        .unwrap();
-        let result = db.store.insert_entry(duplicate_doggo_entry).await;
+            let duplicate_doggo_entry = StorageEntry::new(
+                first_entry.entry_signed(),
+                first_entry.operation_encoded().unwrap(),
+            )
+            .unwrap();
 
-        assert_eq!(
-            result.unwrap_err().to_string(),
-            "Error occured during `EntryStorage` request in storage provider: error returned from \
-            database: UNIQUE constraint failed: entries.author, entries.log_id, entries.seq_num"
-        )
+            let result = db.store.insert_entry(duplicate_doggo_entry).await;
+            assert!(result.is_err());
```

Comment:
Returned error strings are different depending on the used database, so we need to just check via `is_err`